onnxruntime/core/session/inference_session.h

```diff
@@ -100,6 +100,61 @@ class InferenceSession {
   explicit InferenceSession(const SessionOptions& session_options,
                             logging::LoggingManager* logging_manager = nullptr);
 
+  /**
+    Create a new InferenceSession
```

Comment:
>reate a new InferenceSession [](start = 5, length = 28)

I think there is a couple of potential usability issues with these:
- This creates a mix of Load() and construction.
- Now it has to throw and it creates a mix of error handling on the user part which propagates to our other APIs including Python.

I would make these Load() overloads combined with a default construction of the Session